// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// This API is experimental and under-development.

syntax = "proto3";

package sax.server.custom;

import "saxml/protobuf/common.proto";

// Usage:
//
//   custom_req = CustomRequest(model_key="lm1t",
//                             request="some text",
//                             method_name="method_name",
//                             )
//   custom_response = custom_service.Custom(custom_req)
//   print(custom_response.response)
//

message CustomRequest {
  string model_key = 1;
  bytes request = 2;
  .sax.ExtraInputs extra_inputs = 3;
  string method_name = 4;
}

message CustomResponse {
  bytes response = 1;
}

service CustomService {
  // Returns result of CustomRequest.
  rpc Custom(CustomRequest) returns (CustomResponse);
}
